test/e2e/kubectl/kubectl.go

```diff
@@ -2069,6 +2069,103 @@ metadata:
 			gomega.Expect(outBuiltIn).To(gomega.Equal(outStatusSubresource))
 		})
 	})
+
+	ginkgo.Describe("Kubectl rollout", func() {
+		deploymentYaml := `
+		{
+			"apiVersion": "apps/v1",
+			"kind": "Deployment",
+			"metadata": {
+			  "name": "nginx-deployment"
+			},
+			"spec": {
+			  "replicas": 1,
+			  "selector": {
+				"matchLabels": {
+				  "test": "rollout"
+				}
+			  },
+			  "template": {
+				"metadata": {
+				  "labels": {
+					"test": "rollout"
+				  }
+				},
+				"spec": {
+				  "containers": [
+					{
+					  "name": "nginx-container",
+					  "image": "nginx:latest"
+					}
+				  ]
+				}
+			  }
+			}
+		}	  
+`
+		ginkgo.AfterEach(func() {
+			cleanupKubectlInputs(deploymentYaml, ns, "test=rollout")
+		})
+		ginkgo.It("update deployment env and rollback", func(ctx context.Context) {
+			// create deployment
+			if _, err := e2ekubectl.RunKubectlInput(ns, deploymentYaml, "create", "-f", "-"); err != nil {
+				framework.Failf("kubectl failed to create deployment")
+			}
+			if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
+				deployment, err := c.AppsV1().Deployments(ns).Get(ctx, "nginx-deployment", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				readyReplicas := deployment.Status.ReadyReplicas
+				desiredReplicas := deployment.Spec.Replicas
+				return readyReplicas == *desiredReplicas, nil
+			}); err != nil {
+				framework.Failf("created deployment not ready")
+			}
+
+			// do a small update
+			if _, err := e2ekubectl.RunKubectl(ns, "set", "env", "deployment/nginx-deployment", "foo=bar"); err != nil {
+				framework.Failf("kubectl failed set env for deployment")
+			}
+			// wait for env to be set
+			if err := wait.PollUntilContextCancel(ctx, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
\+ here. You can check for the complete deployment and then check for the presence of the env variable